In [1]:
from math import inf

Варіант №7:

Завдання 3

- Виконати первинну обробку фінансових показників діяльності
підприємства за ІІ квартал;

Завдання 4

- Знайти значення функції належності лінгвістичної змінної «ризик
банкрутства підприємства» за ІІ квартал;

| Назва показника                                                  	| Позначення показника 	| I квартал 	| II квартал 	|
|------------------------------------------------------------------	|----------------------	|-----------	|------------	|
| Коефіцієнт автономії                                             	| $X_1$                	| 0.4       	| 0.15       	|
| Коефіцієнт забезпечення оборотних <br> активів власними засобами 	| $X_2$                	| 0.51      	| 0.32       	|
| Коефіцієнт швидкої ліквідності                                   	| $X_3$                	| 1.3       	| 0.9        	|
| Коефіцієнт абсолютної ліквідності                                	| $X_4$                	| 0.35      	| 0.7        	|
| Коефіцієнт оборотності активів                                   	| $X_5$                	| 0.73      	| 1.0        	|
| Коефіцієнт рентабельності капіталу                               	| $X_6$                	| 0.18      	| 0.3        	|

In [2]:
# Спостережуване значення показника
X1 = 0.15
X2 = 0.32
X3 = 0.9  
X4 = 0.7
X5 = 1.0
X6 = 0.3

$B_{i1}$ – «дуже мале значення показника $X_i$»; <br>
$B_{i2}$ – «мале значення показника $X_i$»; <br>
$B_{i3}$ – «середнє значення показника $X_i$»; <br>
$B_{i4}$ – «велике значення показника $X_i$»; <br>
$B_{i5}$ – «дуже велике значення показника $X_i$». <br>

$X_1$, $B_{11}$ = (0; 0; 0.1; 0.2) <br>
$X_2$, $B_{24}$ = (0.3; 0.35; 0.45; 0.5) <br>
$X_3$, $B_{33}$ = (0.7; 0.8; 0.9; 1) <br>
$X_4$, $B_{45}$ = (0.5; 0.6; $\infty$; $\infty$) <br>
$X_5$, $B_{55}$ = (0.5;0.8; $\infty$; $\infty$) <br>
$X_6$, $B_{64}$ = (0.06; 0.1; 0.225; 0.4) <br>

Терм-множина показника

In [3]:
B11 = (0, 0, 0.1, 0.2)
B24 = (0.3, 0.35, 0.45, 0.5)
B33 = (0.7, 0.8, 0.9, 1)
B45 = (0.5, 0.6, inf, inf)
B55 = (0.5, 0.8, inf, inf)
B64 = (0.06, 0.1, 0.225, 0.4) 

In [4]:
# Формула функції належності узагальненого трапецєподібного нечіткого інтервалу 
def mu(x, arr):
    a1, a2, a3, a4 = arr
    if x < a1:
        return 0
    if a1 <= x < a2:
        return (x - a1) / (a2 - a1)
    if a2 < x <= a3:
        return 1
    if a3 < x <= a4:
        return (x - a4) / (a3 - a4)
    if x > a4:
        return 0

Значення функції належності

In [5]:
mu_11 = mu(X1, B11)
mu_24 = mu(X2, B24)
mu_33 = mu(X3, B33)
mu_45 = mu(X4, B45)
mu_55 = mu(X5, B55)
mu_64 = mu(X6, B64)

Вагові коефіцієнти термів $G_k$

In [6]:
p5 = 1/6 * mu_11
p4 = 0
p3 = 1/6 * mu_33
p2 = 1/6 * (mu_24 + mu_64)
p1 = 1/6 * (mu_45 + mu_55)

Значення лінгвістичної змінної g на k - му проміжку, $g_k$ = $p_k$ $\bar{g_k}$

In [7]:
g1 = p1 * 0.875
g2 = p2 * 0.7
g3 = p3 * 0.5
g4 = p4 * 0.3
g5 = p5 * 0.125

In [8]:
g = g1 + g2 + g3 + g4 + g5
print(g)

0.49875


$\mu_3$(0.49875) = 1 $\mid_{g=0.49875}$ = 1.

Таким чином, у ІІ кварталі підприємство перебувало у стані

$G_3$ ($\mu_3$ = 1) «ризик банкрутства середній».